hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/EventJournalUpdaterRecordStoreMutationObserver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.recordstore;
+
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.journal.MapEventJournal;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.nio.serialization.Data;
+
+public class EventJournalUpdaterRecordStoreMutationObserver implements RecordStoreMutationObserver {
+    private final MapEventJournal eventJournal;
+    private final MapContainer mapContainer;
+    private final int partitionId;
+
+    public EventJournalUpdaterRecordStoreMutationObserver(MapEventJournal eventJournal, MapContainer mapContainer,
+                                                          int partitionId) {
+        this.eventJournal = eventJournal;
+        this.mapContainer = mapContainer;
+        this.partitionId = partitionId;
+    }
+
+    @Override
+    public void onClear() {
+        // NOP
+    }
+
+    @Override
+    public void onPutRecord(Data key, Record record) {
+        eventJournal.writeAddEvent(mapContainer.getEventJournalConfig(),
+                mapContainer.getObjectNamespace(), partitionId, record.getKey(), record.getValue());
+    }
+
+    @Override
+    public void onReplicationPutRecord(Data key, Record record) {
+        // NOP
+    }
+
+    @Override
+    public void onUpdateRecord(Data key, Record record, Object newValue) {
+        eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
```

Comment:
Minor: you can pass in the config and the namespace via constructor rather than looking it up on each event.